src/plugins/visualizations/server/embeddable/make_visualize_embeddable_factory.ts

```diff
@@ -84,26 +89,37 @@ const byValueRemoveMarkdownLessFromTSVB = (state: SerializableRecord) => {
   };
 };
 
-export const visualizeEmbeddableFactory = (): EmbeddableRegistryDefinition => {
-  return {
-    id: 'visualization',
-    migrations: {
-      // These migrations are run in 7.13.1 for `by value` panels because the 7.13 release window was missed.
-      '7.13.1': (state) =>
-        flow(
-          byValueAddSupportOfDualIndexSelectionModeInTSVB,
-          byValueHideTSVBLastValueIndicator,
-          byValueRemoveDefaultIndexPatternAndTimeFieldFromTSVBModel
-        )(state),
-      '7.14.0': (state) =>
-        flow(
-          byValueAddEmptyValueColorRule,
-          byValueMigrateVislibPie,
-          byValueMigrateTagcloud,
-          byValueAddDropLastBucketIntoTSVBModel
-        )(state),
-      '7.17.0': (state) => flow(byValueAddDropLastBucketIntoTSVBModel714Above)(state),
-      '8.0.0': (state) => flow(byValueRemoveMarkdownLessFromTSVB)(state),
-    },
+const getEmbeddedVisualizationSearchSourceMigrations = (
+  searchSourceMigrations: MigrateFunctionsObject
+) =>
+  mapValues(searchSourceMigrations, (migrate) =>
+    getApplyMigrationWithinObject(migrate, 'data.searchSource')
```

Comment:
This seems to be wrong - I guess it should be `visState.data.searchSource`?

Reply:
Yes, that was the issue!